sdk/eventhub/eventhubs-checkpointstore-table/test/tables-checkpointstore.spec.ts

```diff
@@ -0,0 +1,556 @@
+    });
+
+    describe("listOwnership", function() {
+      it("listOwnership should return an empty array", async function(): Promise<void> {
+        const checkpointStore = new TableCheckpointStore(client);
+        const listOwnership = await checkpointStore.listOwnership(
+          "test.servicebus.windows.net",
+          "testHub",
+          "testConsumerGroup"
+        );
+        should.equal(listOwnership.length, 0);
+      });
+    });
+
+    describe("updateOwnership", function() {
+      it("forwards errors", async () => {
+        const checkpointStore = new TableCheckpointStore(client);
+        const eventHubProperties = {
+          fullyQualifiedNamespace: "brown.servicebus.windows.net",
+          eventHubName: "testEventHub",
+          consumerGroup: "testConsumerGroup"
+        };
+        // now let's induce a bad failure (removing the table)
+        await serviceClient.deleteTable(table_name);
+
+        // Create the checkpoint to add.
+        const checkpoint: Checkpoint = {
+          consumerGroup: eventHubProperties.consumerGroup,
+          eventHubName: eventHubProperties.eventHubName,
+          fullyQualifiedNamespace: eventHubProperties.fullyQualifiedNamespace,
+          offset: 0,
+          partitionId: "0",
+          sequenceNumber: 1
+        };
+        try {
+          await checkpointStore.updateCheckpoint(checkpoint);
+          throw new Error("Failed");
+        } catch (err) {
+          err.message.should.not.equal("Failed");
+        }
+      });
+    });
+
+    describe("Runs tests on a populated table", function() {
+      beforeEach("creating table", async () => {
+        table_name = `table${new Date().getTime()}`;
+        client = new TableClient(
+          `https://${service.storageAccountName}.table.core.windows.net`,
+          table_name,
+          credential
+        );
+        await serviceClient.createTable(table_name);
+        const namespaceArray = [
+          "red.servicebus.windows.net",
+          "blue.servicebus.windows.net",
+          "green.servicebus.windows.net"
+        ];
+        const eventHubArray = ["redHub", "blueHub", "greenHub"];
+        const consumerConst = "$default";
+        /* Checkpoint */
+        const checkpoint_entity: CheckpointEntity = {
+          partitionKey: "",
+          rowKey: "",
+          sequencenumber: 0,
+          offset: 0
+        };
+        for (let i = 0; i < 3; ++i) {
+          checkpoint_entity.sequencenumber = 100 + i;
+          checkpoint_entity.rowKey = i.toString();
+          checkpoint_entity.partitionKey = `${namespaceArray[i]} ${eventHubArray[i]} ${consumerConst} Checkpoint`;
+          checkpoint_entity.offset = 1023 + i;
+          await client.createEntity(checkpoint_entity);
+        }
+
+        /* Ownership */
+        const ownership_entity: PartitionOwnershipEntity = {
+          partitionKey: "",
+          rowKey: "",
+          ownerid: ""
+        };
```

Comment:
Nit: you can move this into the for loop below and assign values directly to the fields rather than initializing them as empty strings first.

Reply:
nice. realized it's neat and makes more sense that way